internal/component/prometheus/operator/common/component.go

```diff
@@ -62,55 +62,70 @@ func (c *Component) CurrentHealth() component.Health {
 	return c.health
 }
 
+type cancelHandler struct {
+	mut            sync.RWMutex
```

Comment:
Is it really necessary to use a mutex? I thought only one case in the switch statement can be active at a given time.

Reply:
I did this out of an abundance of caution and to ensure race didnt trigger. It may not be needed.